util/util.go

```diff
@@ -933,6 +960,18 @@ func GenerateDiskConfig(path string) (*DiskConfig, error) {
 	if _, err := nsExec.Execute("ls", []string{filePath}); err == nil {
 		return nil, fmt.Errorf("disk cfg on %v exists, cannot override", filePath)
 	}
+
+	// Cleaning up disk cfg file and replica subdirectory if anything
+	// goes wrong
+	defer func() {
+		if err != nil {
+			if derr := DeleteDiskPathReplicaSubdirectoryAndDiskCfgFile(path); derr != nil {
```

Comment:
nsExec can be passed in as a parameter for the function, so we don't need to create it again.